src/github.com/kelseyhightower/confd/backends/config-service/client.go

```diff
@@ -66,28 +69,46 @@ func (c *Client) GetValues(keys []string) (map[string]string, error) {
 
 
 		for _, dynamicBucket := range dynamicBuckets {
-			val := dynamicBucket.GetKeys()[key]
-			if val == nil {
-				continue;
-			}
-			valType := reflect.TypeOf(val).Kind()
-			if valType == reflect.Slice {
-				data, err := ffjson.Marshal(val)
-				if err != nil {
-				    log.Error("Failed decoding from JSON")
-				} else {
-					vars[key] = string(data[:])
+			var requestedKeys []string
+
+			if key == "*" {
+				//when key is "*" get all keys in a bucket,
+				bucketKeys := dynamicBucket.GetKeys()
+				requestedKeys = []string{}
+				for k, _ := range bucketKeys {
```

Comment:
can this loop be replaced by a simple append(requestedKeys, dynamicBucket.GetKeys()...)

Reply:
dynamicBucket.GetKeys() returns array of interface{}, converting this to string in loop.